src/command.rs

```diff
@@ -26,28 +27,57 @@ use crate::{generator, generator::Language};
 use std::path::PathBuf;
 
 use async_recursion::async_recursion;
-use eyre::{ensure, Context, ContextCompat};
+use displaydoc::Display;
 use semver::{Version, VersionReq};
 use std::{env, path::Path, sync::Arc};
+use thiserror::Error;
 
 const INITIAL_VERSION: Version = Version::new(0, 1, 0);
 
-/// Initializes the project
-pub async fn init(kind: PackageType, name: Option<PackageName>) -> eyre::Result<()> {
-    ensure!(
-        !Manifest::exists().await?,
-        "Cannot re-initialize an existing project"
-    );
+#[derive(Error, Display, Debug)]
+#[non_exhaustive]
+#[allow(missing_docs)]
+pub enum InitError {
+    /// a manifest file was found, project is already initialized
+    AlreadyInitialized,
+    /// invalid package name
+    InvalidName(#[from] InvalidPackageName),
+    /// io error: {message}
+    Io {
+        message: String,
+        source: std::io::Error,
+    },
+    /// failed to create manifest file
+    ManifestWrite(#[from] manifest::WriteError),
+    /// failed to create package store
+    PackageStore(#[from] package::CreateStoreError),
+    /// {0}
```

Comment:
is this a meaningful doc comment

Reply:
Not at all. It's just there to tell displaydoc how to render the Other variant (ie, transparently). I could use `#[displaydoc("...")]` to make it possible to additionally include a non-display doc-comment, but I'm not sure what I'd put there.